src/strings.cpp

```diff
@@ -381,7 +382,7 @@ static char *FormatZerofillNumber(char *buff, int64 number, int64 count, const c
 
 static char *FormatHexNumber(char *buff, uint64 number, const char *last)
 {
-	return buff + seprintf(buff, last, "0x" OTTD_PRINTFHEX64, number);
+	return strecpy(buff, fmt::format("0x{:X}", number).c_str(), last);
```

Comment:
The more efficient way would have been:
```
       auto [ptr, ec] = std::to_chars(buff, const_cast<char *>(last), number, 16);
       if (ec == std::errc()) buff = ptr;

       /* std::to_chars does not zero-terminate. */
       *buff = '\0';
       return buff;
```
However, that fails to build on OS X because it is not available until OS X 10.15.
So I have reverted to a simpler, though slightly less efficient manner. Which should be fine as it's only used in NewGRF error messages right now.